Added get_culvert

jacob-a-brown opened the pull request:
Added the method get_culvert to parserasgeo
to get a specific culvert based on its id, river, and reach from a model's geo file

Commit:
Added get_culvert

Added the method get_culvert to parserasgeo
to get a specific culvert based on its id, river, and reach from a model's geo file

mikebannis commented on parserasgeo/prg.py:
```diff
@@ -150,6 +153,46 @@ def return_xs(self, xs_id, river, reach, strip=False, rnd=False, digits=0):
                     return item
         raise CrossSectionNotFound
+        
+    def return_culvert(self, culvert_id, river, reach, strip=False, rnd=False, digits=0):
```
return_culvert() and return_xs() are identical except for the type of feature (class) used to compare against in new line 181 (      if isinstance(item, Culvert): ).  Create method _return_node() which also accepts a feature type (CrossSection, Culvert, etc) and is called by wrappers return_culvert() and return_xs().

jacob-a-brown committed:
Semi-private extract and return methods

Added semi-private methods to extract and return nodes
from a geo file. These changes were made so that code does
not need to be copy-pasted

mikebannis commented on parserasgeo/prg.py:
```diff
@@ -198,6 +179,81 @@ def is_xs_duplicate(self, xs_id):
+
+        for item in self.geo_list:
+            if isinstance(item, Culvert):
+                test_river = item.river
```
test_river and test_reach are never set if item is not a Culvert. Was this tested before the PR?

jacob-a-brown committed:
Addressed comments

Changed line 222 to be "if isinstance(item, node_type)" so that this
works for all node types

jacob-a-brown committed:
Cleaned protected methods

Cleaned the protected methods so that they
work properly. Since an instance is not passed, but the class
is, used class.__name__ instead of isinstance
